portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,10 +34,61 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
   private String name;
+  private ArrayList<String> comments = new ArrayList<String>();
+  private List<Task> tasks = new ArrayList<>();
+  private static final String TEXT_INPUT = "text-input";
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String name = "Bless";
+    Query query = new Query("Task").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    for (Entity entity : results.asIterable()) {
+      long id = entity.getKey().getId();
+      String title = (String) entity.getProperty("title");
+      System.out.println("title" + title);
+      long timestamp = (long) entity.getProperty("timestamp");
+      System.out.println("time" + timestamp);
+      Task task = new Task(id, title, timestamp);
+      tasks.add(task);
```

Comment:
You keep adding to the same list, which probably results in the list growing indefinitely. Could you create a new list inside doGet and get rid of the `tasks` field? I think that'd fix your issue